datafusion/core/src/execution/context.rs

```diff
@@ -1247,16 +1250,26 @@ impl SessionState {
         rules.push(Arc::new(LimitPushDown::new()));
         rules.push(Arc::new(SingleDistinctToGroupBy::new()));
 
+        let mut physical_optimizers: Vec<Arc<dyn PhysicalOptimizerRule + Sync + Send>> = vec![
+            Arc::new(AggregateStatistics::new()),
+            Arc::new(HashBuildProbeOrder::new()),
+        ];
+        if config.config_options.get_bool(OPT_COALESCE_BATCHES) {
```

Comment:
I wonder if it would be helpful to add a test to ensure the option to disable coalsce'ing batches doesn't get broken in the future

Reply:
Sounds good. I will get to this in the next day or two.